lua/cfc_prop_restoration/core/sv/sv_init.lua

```diff
@@ -205,15 +269,21 @@ timer.Create( "CFC_Restoration_Think", 5, 0, function()
     if time >= nextSave and table.IsEmpty( queue ) then
         logger:info( "Autosaving player props" )
 
+        getAllPlayerProps()
+
         for _, ply in pairs( player.GetHumans() ) do
 
+            local propVelocities = getPropVelocities( ply )
+
             local success, props = xpcall( ADInterface.copy, notifyOnError( ply ), ply )
             success = success and props
 
             if success and not table.IsEmpty( props ) and props ~= nil then
                 propData[ply:SteamID()] = props
                 addPropDataToQueue( ply, props )
             end
+
+            restorePropVelocities( propVelocities )
```

Comment:
I think `getAllPlayerProps` should return the table of player props, and then you pass a table of each player's props into `getPropVelocities`

Something like this:
```suggestion
        local playerProps = getAllPlayerProps()

        for _, ply in pairs( player.GetHumans() ) do

            local propVelocities = getPropVelocities( playerProps[ply] )

            local success, props = xpcall( ADInterface.copy, notifyOnError( ply ), ply )
            success = success and props

            if success and not table.IsEmpty( props ) and props ~= nil then
                propData[ply:SteamID()] = props
                addPropDataToQueue( ply, props )
            end

            restorePropVelocities( propVelocities )
```